crates/verify/src/verify.rs

```diff
@@ -321,7 +322,14 @@ impl VerifyArgs {
                 let profiles = entry
                     .artifacts
                     .get(&contract.name)
-                    .and_then(|artifacts| artifacts.get(&version))
+                    .and_then(|artifacts| {
+                        let mut cached_artifacts = artifacts.get(&version);
+                        if cached_artifacts.is_none() && version.build != BuildMetadata::EMPTY {
+                            version.build = BuildMetadata::EMPTY;
+                            cached_artifacts = artifacts.get(&version);
+                        }
+                        cached_artifacts
```

Comment:
would you mind adding a clarifying comment here explaining under what circumstances this could happen?

Reply:
Thanks!